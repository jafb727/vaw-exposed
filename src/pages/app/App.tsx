/**
 * @author Jose A Felix
 * @name App
 * @description Main application component
 */

/** -------------------------------------------------------- */

/** @import Hooks */
import { useState } from "react";

/** @import Utilities */
import { JSX } from "react";

/** @import Assets */
import reactLogo from "../../assets/images/react.svg";
import viteLogo from "../../assets/images/vite.svg";

/** @import Stylesheets */
import "./app.css";

/** -------------------------------------------------------- */

/**
 * @name App
 * @description Main application component
 * @returns {JSX.Element}
 */
const App = (): JSX.Element => {
   /** State */
   const [count, setCount] = useState(0);

   /** ------------------------------ */

   /**
    * @name countClicks
    * @description Increments counter according to user clicks
    * @returns {void}
    */
   const countClicks = (): void => {
      setCount((count) => {
         return count + 1;
      });
   };

   /** ------------------------------ */

   const properties = {
      "data-testid": "app-page",
   };

   /** ------------------------------ */

   return (
      <div {...properties}>
         <div>
            <a href="https://vite.dev" target="_blank" rel="noreferrer">
               <img src={viteLogo} className="logo" alt="Vite logo" />
            </a>
            <a href="https://react.dev" target="_blank" rel="noreferrer">
               <img src={reactLogo} className="logo react" alt="React logo" />
            </a>
         </div>
         <h1>Vite + React</h1>
         <div className="card">
            <button onClick={countClicks}>count is {count}</button>
            <p>
               Edit <code>src/App.tsx</code> and save to test HMR
            </p>
         </div>
         <p className="read-the-docs">
            Click on the Vite and React logos to learn more
         </p>
      </div>
   );
};
/** -------------------------------------------------------- */

/** @exports @default Component */
export default App;
