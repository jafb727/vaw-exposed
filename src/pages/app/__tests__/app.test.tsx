/**
 * @author Jose A Felix
 * @name app.test
 * @description Main application component unit testing file
 */

/** -------------------------------------------------------- */

/** @import Components */
import App from "..";

/** @import Utilities */
import { renderComponent } from "../../../test/setupTests";
import userEvent from "@testing-library/user-event";

/** -------------------------------------------------------- */

describe("App", () => {
   it("Renders ok", () => {
      /** Component context mounting */
      const component = renderComponent(<App />);

      /** Main assertion */
      expect(component).not.toBe(null);
   });

   it("Renders main component wrapper", () => {
      /** Component context mounting */
      const { getByTestId } = renderComponent(<App />);

      /** Main assertion */
      const component = getByTestId("app-page");
      expect(component).toBeInstanceOf(HTMLElement);
   });

   it("Increases count when clicking button", async () => {
      /** Component context mounting */
      const { getByTestId, findByTestId } = renderComponent(<App />);

      /** Triggering user interaction */
      const button = getByTestId("counter");
      await userEvent.click(button);

      /** Main assertion */
      const _button = await findByTestId("counter");
      expect(_button.textContent).toContain("1");
   });

   it("Matches snapshot", () => {
      /** Component context mounting */
      const { getByTestId } = renderComponent(<App />);

      /** Main assertion */
      const component = getByTestId("app-page");
      expect(component).toMatchSnapshot();
   });
});
