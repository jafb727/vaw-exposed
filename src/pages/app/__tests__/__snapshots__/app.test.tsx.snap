// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App Matches snapshot 1`] = `
<div
  data-testid="app-page"
>
  <div>
    <a
      href="https://vite.dev"
      rel="noreferrer"
      target="_blank"
    >
      <img
        alt="Vite logo"
        class="logo"
        src="mock-svg"
      />
    </a>
    <a
      href="https://react.dev"
      rel="noreferrer"
      target="_blank"
    >
      <img
        alt="React logo"
        class="logo react"
        src="mock-svg"
      />
    </a>
  </div>
  <h1>
    Vite + React
  </h1>
  <div
    class="card"
  >
    <button
      data-testid="counter"
    >
      count is 
      0
    </button>
    <p>
      Edit 
      <code>
        src/App.tsx
      </code>
       and save to test HMR
    </p>
  </div>
  <p
    class="read-the-docs"
  >
    Click on the Vite and React logos to learn more
  </p>
</div>
`;
